test/test_tools.rb

```diff
@@ -146,11 +146,52 @@ def test_change_rpath_fat
   end
 
   def test_add_rpath
-    pass
+    groups = SINGLE_ARCHES.map do |arch|
+      ["hello.bin", "hello_actual.bin"].map do |fn|
+        fixture(arch, fn)
+      end
+    end
+
+    groups.each do |filename, actual|
+      FileUtils.cp filename, actual
+      MachO::Tools.add_rpath(actual, "/foo/bar/baz")
+
+      original = MachO::MachOFile.new(filename)
+      modified = MachO::MachOFile.new(actual)
+
+      assert_operator modified.ncmds, :>, original.ncmds
+      assert_operator modified.sizeofcmds, :>, original.sizeofcmds
+      assert_operator modified.rpaths.size, :>, original.rpaths.size
+      refute_includes original.rpaths, "/foo/bar/baz"
+      assert_includes modified.rpaths, "made_up_path"
```

Comment:
Oh, right! This is all about addition so of course the pre-existing RPATH is still included.